src/uwtools/rocoto.py

```diff
@@ -26,6 +48,67 @@ def _add_jobname(tree: dict) -> None:
 
 
 # Public functions
+def realize_rocoto_xml(
+    input_yaml: str,
+    rendered_output: str,
+) -> bool:  # pragma: no cover
+    """
+    Main entry point.
+
+    :param input_yaml: Path to YAML input file.
+    :param input_template: Path to input template file.
+    :param rendered_output: Path to write rendered XML file.
+    :param schema_file: Path to schema file.
+    """
+    values = YAMLConfig(input_yaml)
+    tasks = values["tasks"]
+    if isinstance(tasks, dict):
+        _add_jobname(tasks)
```

Comment:
This chunk of work is repeated in the `write_rocoto_xml`. Maybe it could be pulled out into its own private function?

Reply:
ah good point, thanks!